core/serializers.py

```diff
@@ -0,0 +1,63 @@
+from rest_framework import serializers
+
+from core import models
+
+
+# class PasswordChangeSerializer(serializers.ModelSerializer):
+#     """A serializer for our user profile objects."""
+#
+#     confirm_pwd = serializers.CharField()
+#
+#     class Meta:
+#         model = models.UserProfile
+#         fields = ('id', 'password', 'confirm_pwd')
+
+
+class UserProfileSerializer(serializers.ModelSerializer):
+    """A serializer for our user profile objects."""
+
+    class Meta:
+        model = models.UserProfile
+        fields = ('id', 'username', 'email', 'first_name', 'password', 'mobile_number')
+        extra_kwargs = {'password': {'write_only': True}}
+
+
+    def create(self, validated_data):
+        """Create and return a new user."""
+
+        user = models.UserProfile(
+            username=validated_data['username'],
+            email=validated_data['email'],
+            first_name=validated_data['first_name'],
+            mobile_number=validated_data['mobile_number'],
```

Comment:
is there a chance that `KeyError` can occur.

Reply:
I have read from here about KeyError, I didn't get where is the change of occurrence of KeyError.
https://docs.python.org/3/library/exceptions.html#KeyError

Reply:
e.g if we don't have `mobile_number` in `validated_date`, it will throw `KeyError`

Reply:
This is how validation is working on the api at this moment
<img width="1083" alt="Screen Shot 2019-08-08 at 8 14 40 PM" src="https://user-images.githubusercontent.com/25842457/62715169-64d2dc00-ba19-11e9-8749-760eb120665a.png">
<img width="1077" alt="Screen Shot 2019-08-08 at 8 14 58 PM" src="https://user-images.githubusercontent.com/25842457/62715170-64d2dc00-ba19-11e9-8e58-52ed782bdb78.png">